File: 2025-10/2025-10-30/nthPrime.js
function nthPrime(n) {
  if (n == 1)
    return 2;

  let testedNumber = 1;
  let primesFound = 1;

  while (primesFound < n) {
    testedNumber += 2;
    let isPrime = true;

    for (let divisor = 3; divisor < testedNumber / 2; divisor += 2)
      if (testedNumber % divisor === 0) {
        isPrime = false;
        break;
      }

    if (isPrime)
        primesFound ++;
  }
  
  return testedNumber;
}

console.log(nthPrime(5));
console.log(nthPrime(10));
console.log(nthPrime(16));
console.log(nthPrime(99));
console.log(nthPrime(1000));

console.log(nthPrime(1));
console.log(nthPrime(2));
console.log(nthPrime(3));

// --- TEST SUITE ---

const testsText = `
1. nthPrime(5) should return 11.
2. nthPrime(10) should return 29.
3. nthPrime(16) should return 53.
4. nthPrime(99) should return 523.
5. nthPrime(1000) should return 7919.
`;

const testsRegex = /(?<number>\d+)\.\s(?<functionCall>.+) should return (?<output>.+?)\.?$/gm;
const testData = [...testsText.matchAll(testsRegex).map(match => match.groups)];

function runTests(testData) {
  console.log("--------------------------");
  console.log("🧪Starting Verification...");
  console.log("--------------------------");

  let allPassed = true;
  let failCount = 0;
  
  testData.forEach(test => {
    const functionCallOutput = eval(test.functionCall);
    const testOutput = eval(test.output);
    let comparison;

    if (Array.isArray(testOutput))
      comparison = arraysEqual(functionCallOutput, testOutput);
    else
      comparison = functionCallOutput === testOutput;

    if (comparison) {
      console.log(`${test.number}.✅PASS - Function Call: ${test.functionCall}`);
    }
    else {
      console.log(`${test.number}.❌FAIL - Function Call: ${test.functionCall}\nExpected: ${testOutput}\nGot: ${functionCallOutput}`);
      allPassed = false;
      failCount ++;
    }
  })

  console.log("----------------------------");

  if (allPassed)
    console.log("🎉SUCCESS: All tests PASSED.");
  else
    console.log(`⚠️WARNING: ${failCount}/${testData.length} tests FAILED.`);
}

function arraysEqual(a, b) {
  if (a.length !== b.length)
    return false;

  for (const index in a) {
    if (Array.isArray(a[index])) {
      if (Array.isArray(b[index])) {
        if (!arraysEqual(a[index], b[index]))
          return false;
      }
      else 
        return false;
    }

    else if (a[index] !== b[index])
      return false;
  }

  return true;
}

runTests(testData);

File: 10 October/10-30/nth_prime.py
def nth_prime(n):
    if n == 1:
        return 2

    tested_number = 1
    primes_found = 1

    while primes_found < n:
        tested_number += 2
        is_prime = True

        for divisor in range(3, tested_number // 2, 2):
            if tested_number % divisor == 0:
                is_prime = False
                break

        if is_prime:
            primes_found += 1

    return tested_number

print(nth_prime(5))
print(nth_prime(10))
print(nth_prime(16))
print(nth_prime(99))
print(nth_prime(1000))

print(nth_prime(1))
print(nth_prime(2))
print(nth_prime(3))

# --- TEST SUITE ---

tests_text = '''
1. nth_prime(5) should return 11.
2. nth_prime(10) should return 29.
3. nth_prime(16) should return 53.
4. nth_prime(99) should return 523.
5. nth_prime(1000) should return 7919.
'''

import re

tests_regex = re.compile(r"(?P<number>\d+)\.\s(?P<function_call>.+) should return (?P<output>.+?)\.?$", re.MULTILINE)
test_data = [item.groupdict() for item in re.finditer(tests_regex, tests_text)]

def run_tests(test_data):
    print("--------------------------")
    print("🧪 Starting Verification...")
    print("--------------------------")

    all_passed = True
    fail_count = 0

    for test in test_data:
        function_call_output = eval(test['function_call'])
        test_output = eval(test['output'])
        
        if function_call_output == test_output:
            print(f"{test['number']}.✅ PASS - Function Call:\n{test['function_call']}")
        else:
            print(f"{test['number']}.❌ FAIL - Function Call:\n{test['function_call']}\nExpected:\n{test_output}\nGot:\n{function_call_output}")
            all_passed = False
            fail_count += 1

    print("----------------------------");

    if (all_passed):
        print("🎉 SUCCESS: All tests PASSED.");
    else:
        print(f"⚠️ WARNING: {fail_count}/{len(test_data)} tests FAILED.");

run_tests(test_data)